File: backend/src/main/java/com/devpilot/user/presentation/DevTokenResponse.java
package com.devpilot.user.presentation;

import java.time.Instant;

/** {@code POST /api/v1/dev/token} 응답 (docs/05 §1.4.5). {@code tokenType}은 항상 {@code Bearer}다. */
public record DevTokenResponse(String accessToken, String tokenType, Instant expiresAt) {}

File: app/lib/core/auth/dev_token_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'dev_token_response.freezed.dart';
part 'dev_token_response.g.dart';

/// `DevTokenResponse` of `POST /api/v1/dev/token` (docs/05 §1.4.5).
@freezed
abstract class DevTokenResponse with _$DevTokenResponse {
  const factory DevTokenResponse({
    required String accessToken,
    required String tokenType,
    required DateTime expiresAt,
  }) = _DevTokenResponse;

  factory DevTokenResponse.fromJson(Map<String, Object?> json) => _$DevTokenResponseFromJson(json);
}
